services/appnavstate/api/src/main/kotlin/io/element/android/services/appnavstate/api/AppForegroundStateService.kt

```diff
@@ -18,8 +18,28 @@ interface AppForegroundStateService {
      */
     val isInForeground: StateFlow<Boolean>
 
+    /**
+     * Updates to whether the app is in an active call or not will be emitted here.
+     */
+    val isInCall: StateFlow<Boolean>
+
+    /**
+     * Updates to whether the app is syncing a notification event or not will be emitted here.
+     */
+    val isSyncingNotificationEvent: StateFlow<Boolean>
+
     /**
      * Start observing the foreground state.
      */
-    fun start()
+    fun startObservingForeground()
+
+    /**
+     * Update the in-call state.
+     */
+    fun updateIsInCallState(isInCall: Boolean)
+
+    /**
+     * Update the active state for the syncing notification event flow.
+     */
+    fun updateIsSyncingNotificationEvent(isSyncingNotificationEvent: Boolean)
```

Comment:
It's weird to add all those new methods to the interface `AppForegroundService`.

Would it be possible to have a interface in the module `:libraries:architecture`

```
interface SyncRequester {
    val needSync: StateFlow<Boolean>
}
```

that can be implemented by any module who need the sync to start using `ContributesMultibinding` and then let the `SyncOrchestrator` collect all the states?

It will suit more our architecture patterns and will simplify I think the implementation.

I am trying to poc this.

Reply:
I see a problem which is maybe more important if one day we support multi session, I do not think that this will behave as expected, i.e., I think that if a call start on one session, all the session will resume the sync. I am thinking out loud, it's maybe not the case, but we need to check that. Maybe by providing the sessionId to all those new API to be able to do some post filtering.

Reply:
> I see a problem which is maybe more important if one day we support multi session

I had the same thought, but IMO we should cross that bridge when we come to it, I think we'd just overcomplicate things for no reason if we did the change now.

Reply:
OK